megamek/src/megamek/client/ui/swing/OffBoardTargetOverlay.java

```diff
@@ -288,9 +288,21 @@ private Rectangle generateRectangle(OffBoardDirection direction, Rectangle bound
                 int extraXOffset = GUIP.getShowUnitOverview() ? UnitOverview.getUIWidth() : 0;
                 xPosition = boundingRectangle.x + boundingRectangle.width - WIDE_EDGE_SIZE - EDGE_OFFSET - extraXOffset;
                 yPosition = boundingRectangle.y + (int) (boundingRectangle.height / 2) - (int) (NARROW_EDGE_SIZE / 2);
-                return new Rectangle(xPosition, yPosition, WIDE_EDGE_SIZE, NARROW_EDGE_SIZE); // used to be
-                                                                                              // NARROW_EDGE_SIZE,
-                                                                                              // WIDE_EDGE_SIZE);
+                Rectangle myRectangle = new Rectangle(xPosition, yPosition, WIDE_EDGE_SIZE, NARROW_EDGE_SIZE);  // used to be
+                                                                                                                // NARROW_EDGE_SIZE,
+                                                                                                                // WIDE_EDGE_SIZE);
+
+                // Account for possible floating Unit Display blocking arrow icon
+                if (GUIP.getUnitDisplayEnabled() && GUIP.getUnitDisplayLocaton() == 0) {
+                    // Move arrow inward if either side overlaps with the Unit Display
+                    Rectangle udRectangle = clientgui.getUnitDisplayDialog().getBounds();
+                    if ((myRectangle.x + myRectangle.width + 5 > udRectangle.x && myRectangle.x <= udRectangle.x + udRectangle.width) ||
```

Comment:
Add some parenthesis to prevent ambiguity.

Reply:
Sorry, I popped out to a show.
I was on the fence about adding more parentheses: on the one hand, unambiguous; on the other, longer and more difficult to read.
Luckily the operator precedence is with us this time:
https://pages.cs.wisc.edu/~willb/cs302/java-operator-precedence.pdf

But in the future I'll add the parentheses.